### DISCUSSION TITLE
Allow custom sort order

### DISCUSSION THREAD

**Discussion by m0byn** [ente-io/ente#497] [Enhancements]
**Is your feature request related to a problem? Please describe.**
I just imported my Aegis keys and want to migrate. It is annoying that the sort order is not adjustable. I am used to some keys being on top and also sort other keys in a custom way.

**Describe the solution you'd like**
Allow sorting manually by dragging keys around.

Keep up the good work!!

**Comment by subcidr** [comment]
Would love for some sorting too, have a lot of keys and having to scroll down to the ones i use more are a tiny bit annoying. Grouping and Sorting would be a great feature to add ✨

**Comment by AlphaElwedritsch** [comment]
Yes, please add custom sort order.
Very important and the showstopper at the moment to take the move to ente author.

**Comment by Hudzen10** [comment]
+1
Custom sort order would be handy 
(currently achieving that by simply renaming entries with a prefix - 1. 2. 3. etc), 

Additionally, most-recently-used as a sort order would be a nice feature as well - to keep frequently used ones high in the list. [EDIT: i.e. #212 ]

Also a concept of favourites to have a few pinned to the top regardless of alphabetic/custom sort order might be something to consider as well.
(possibly un-favourite then re-favourite to change the order of the favourites and add last favourited item to bottom of favourites list, as an alternative to having the favourites listed in the same order as the full list)
[EDIT 2: i.e. #134 ]

**Comment by TimFinitor** [comment]
Yes, I think that would be great too. Maybe also the possibility to use preset sorting instead of the classic alphabetical sorting currently

**Comment by ghost** [comment]
+1
It would be cool to be able to sort the entries manually, e.g. via drag and drop or other options.
If you have a lot of entries it quickly becomes confusing and if you need a code quickly it becomes more difficult.

**Comment by bboles** [comment]
There is the ability to pin entries if you right-click on an entry:
![image](https://github.com/user-attachments/assets/abffcf8b-cf6c-4f15-9629-c3e3e5b55f01)
I would also like to be able to sort by most recently used but this gets most of the way there in the meantime.

**Comment by TSHsoft** [comment]
If code can be sorted by drag and drop, then it is more convenient.

**Comment by bvanderark** [comment]
The addition of pinned entries is nice.  I'd like to see an option to always show the pinned entries at the top and then scroll up/down the entries below.

**Comment by brog-io** [comment]
This should be released in the next update 😄

**Reply by nktnet1** [reply]
Update - pull request [#4072](https://github.com/ente-io/ente/pull/4072) was released in [v4.1.5](https://github.com/ente-io/ente/releases/tag/auth-v4.1.5).

![instructions](https://github.com/user-attachments/assets/52e6e862-1a9c-412f-9f6a-a4b62dc23577)

**Comment by brog-io** [comment]
This has been added in the latest Auth update
